B:19591 - Address State Validation For Unsupported States

TevinAdams opened the pull request:
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19591)

## Summary

This merge request introduces additional validation check to the address state field. The selected  state now gets checked to see if it is a supported state. If it is not a supported state then the form requiring the address can not be submitted and a message will display above the state field with details.

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

*In all cases the message described in the testing steps should read  "Moves to this state are not supported at this time".

1. Access the MilMove application as a customer. 
2. Login as a new user and when prompted for an address, try to select HI or AK.
3. You should see a message above the State field and the next button should be disabled.
4. After selecting to a supported state(any state but AK and HI) continue with customer info.
5. Next go to set up shipments. Here wherever address fields, including the optional address fields, try to select AK or HI as a state. There should be a message above the state field and the next button should be disabled.
6. Put in a supported state then continue to through the workflow.
7. Next access office as a SC role.
8. Go to add a shipment.
9. Go through all shipment types attempting to put AK or HI as a state. Should not be able to move on and see a message with detail above the state field.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

## Screenshots
![image](https://github.com/transcom/mymove/assets/136514590/70c59434-800f-4f3b-9a81-bf2e78cebe94)
![image](https://github.com/transcom/mymove/assets/136514590/94e3825c-55e8-4da2-84fe-dc49ed4bdc22)
![image](https://github.com/transcom/mymove/assets/136514590/ae50e841-65a4-437c-bed5-3833c785a2d8)

robot-mymove commented:
<!--
  0 failure: 
  1 warning:  This PR does not ...
  
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR does not include changes to unit tests, even though it affects app code.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 21193b1312c8dd41d8c98fb6f9ecd08ddfeb1970
</p>

cameroncaci reviewed (approved):
Love the way this was implemented. Passed all AC on this end

traskowskycaci reviewed (commented):
Functionality works great! Could you add in a few small tests for this too?

TevinAdams commented:
> Functionality works great! Could you add in a few small tests for this too?

@traskowskycaci @cameroncaci  Done. Added 2 tests total. One to Backup address and one to ResidentialForm. All tests for this are pretty much the same so I just added 2.

traskowskycaci reviewed (approved):
Nice work!

danieljordan-caci reviewed (changes requested):
You'll also need to look at `CreateCustomerForm.jsx` for the service counselor

Was putting in this comment when you merged

TevinAdams commented:
> You'll also need to look at `CreateCustomerForm.jsx` for the service counselor
> 
> Was putting in this comment when you merged

After following up with @danieljordan-caci  the CreateCustomerForm was covered by this implementation.

danieljordan-caci commented:
> > You'll also need to look at `CreateCustomerForm.jsx` for the service counselor
> > Was putting in this comment when you merged
> 
> After following up with @danieljordan-caci the CreateCustomerForm was covered by this implementation.

You'll need to remove the `statesList` variable and use the one in `constants.js` - that's the only change I can see. Then ofcourse add tests.

TevinAdams commented:
> > > You'll also need to look at `CreateCustomerForm.jsx` for the service counselor
> > > Was putting in this comment when you merged
> > 
> > 
> > After following up with @danieljordan-caci the CreateCustomerForm was covered by this implementation.
> 
> You'll need to remove the `statesList` variable and use the one in `constants.js` - that's the only change I can see. Then ofcourse add tests.

@danieljordan-caci  Done
